src/27-ipfs-at-the-vatican/index.md

```diff
@@ -0,0 +1,158 @@
+
+Peer to peer tools are the right tech to apply here. With the demo we set up, each node on the network holds its own copy of the annotations. The user of that node decides which streams of annotations they will publish to and subscribe from. Groups of people can share their annotations as they see fit. Third parties like the library can follow or ignore public streams, choosing which content to keep a copy of.
+
+This is an example of the permissionless web — where nobody needs permission to create content or share it. There is no annotation server, so there is no gatekeeper deciding who gets to create annotations; everyone is free to create content as they see fit.
+
+This peer-to-peer approach also gives us an example of the role for cultural institutions on the decentralized web. If each person or group is holding their own copies of the data they deem important, you inevitably need someone to coordinate the effort to ensure ongoing access to those data, to ensure that people will find/discover the content, and to ensure that the content is preserved for the long term. This is exactly the role that Libraries, Archives and Museums exist to fill.
+
+In the demo, we showed annotations synchronizing between the two main IIIF viewers -- Mirador and Universal Viewer. We pointed the viewers at the same IIIF manifest and configured them to use a pre-selected pubsub channel. When a user adds an annotation in Mirador, it publishes the annotation to the pubsub channel and Universal Viewer displays the annotation on its copy of the image.
+
+TODO: VIDEO* Present the demo (screen capture)
+
+## The Tech: IPFS, CRDTs and pubsub
+
+From a tech perspective, the first demo was easy — add all the content to IPFS and replace all the HTTP links with links to IPFS gateways. We use gateway links for backwards compatibility — if a browser that is aware of IPFS (i.e. with a web extension installed) it will intercept those gateway requests and resolve them via IPFS but a browser that doesn't speak IPFS will still work because it just sees the IPFS gateway as a regular HTTP server.
+
+The second demo is where the tech gets interesting. It shows most of the building blocks you need to build a distributed, serverless application. It involves four parts
+
+- IPFS - a way of persisting the data (i.e. the history of annotations) and replicating it between peers
+- pubsub - a means of declaring named _channels_ or _topics_ that participants can _publish_ updates to and _subscribe_ to receive updates. This is how nodes tell each other when they have modified the data.
+- CRDTs - a pattern for formatting your data/messages to support real-time collaborative editing. It allows you to ensure that everyone will see the same info ordered in the same way, even if the info arrived on each machine in a different order.
```

Comment:
Maybe "CRDTs" => "CRDT, *Conflict-free Replicated Data Type*" -- we don't decipher that acronym anywhere and I think it helps understand the significance of CRDTs.